File: Client/Filters/SessionAspect.vb
Public Class SessionAspect
    Inherits ActionFilterAttribute

    Public Overrides Sub OnActionExecuting(context As ActionExecutingContext)
        Dim session As HttpSessionStateBase = context.HttpContext.Session
        Dim sessionValue As Object = session("ActivePerson")
        Dim sessionJson As String = If(TryCast(sessionValue, String), String.Empty)

        If String.IsNullOrEmpty(sessionJson) Then
            context.Result = New RedirectResult("~/Authentication/LogIn")
        End If
    End Sub
End Class

File: MVC/Filters/SessionAspect.cs
﻿using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Filters;

namespace PurchasingSystem.Web.Filters
{
    public class SessionAspect : ActionFilterAttribute
    {
        public override void OnActionExecuting(ActionExecutingContext context)
        {
            var sessionJson = context.HttpContext.Session.GetString("ActivePerson");
            if (string.IsNullOrEmpty(sessionJson))
            {
                context.Result = new RedirectResult("~/Authentication/LogIn");
            }
        }
    }
}
